### DISCUSSION TITLE
New GPU Cloud Integration with Hyperstack!

### DISCUSSION THREAD

**Discussion by williamstein** [sagemathinc/cocalc#7484] [Announcements]
If you're GPU's on CoCalc, there's an entirely new cloud option that you should see which is [Hyperstack](https://www.hyperstack.cloud/):

<img width="348" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/70ef02da-35b5-4a7f-a3e8-c4a9f51dd617">

Once you select Hyperstack after starting to [create a compute server](https://doc.cocalc.com/compute_server.html), then click the A100 tag, you'll see this:

<img width="1379" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/61ef50f8-7590-4561-9b99-e053b2ab2f81">

Note that for $3.60/hour you get an 80GB A100, and these are all standard instances. You can also see that at least right now many are available.  Everything else works very similar to Google cloud, except that:

- startup time is slower -- definitely expect about 5-10 minutes from when you click "Start" until you can use the compute server.  However, it's very likely to work, unlike Google cloud GPU's (especially spot instances).  Google cloud is extremely good for CPU, but for GPU it's not as good.
- Many of the server configurations have over 500GB of very fast local ephemeral disk, in case you need that for scratch.  It's ephemeral, goes away when you stop the server.
- The local disk on the server should be as fast or faster than google cloud, but cheaper.
- All network usage is free, whereas egress from google cloud is quite expensive.
- There's a different range of GPU's.  Sometimes there are a lot of H100's but in the middle of the day Wednesday, there aren't.  Yesterday there were dozens of them.
- By default only a Python (Anaconda) image and an Ollama image are visible, since they are small.  When you select the python  image, you'll likely have to type `conda install ...` in a terminal to install some packages you need.   If you click the "Advanced" checkbox when selecting an image, you can select from the full range of images.  However, the first startup time for your server maybe be MUCH slower for big images (e.g., think "20-30 minutes" for the huge colab image).  Starting the server a second time is fast again.  

<img width="891" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/19cdaa95-d9c4-4dc8-beed-1fc69e24cf8a">

- Live disk enlarging does work.   However, there's a limit on Hyperstack where you can enlarge the disk at most 25 times.

**Comment by mforbes** [comment]
@williamstein Have you explored Jetstream2? (https://jetstream-cloud.org/).  Part of NSF's Accelerate program, and apparently fairly easy for researchers to get access to.

**Reply by williamstein** [reply]
> Have you explored Jetstream2?

I haven't.  I don't think there is any way I could get access to it anymore, since I left academia.
